source/opencascade/utilities.cc

```diff
@@ -462,6 +462,112 @@ namespace OpenCASCADE
     return out_shape;
   }
 
+
+
+  template<int spacedim>
+  std::vector<TopoDS_Edge> create_curves_from_triangulation_boundary
+  (const Triangulation<2, spacedim> &triangulation,
+   const Mapping<2, spacedim> &mapping)
+
+  {
+    // store maps from global vertex index to pairs of global face   indices
+    //        and from global face   index to pairs of global vertex indices
```

Comment:
why pairs? Because each vertex is a part of (at most) two boundary faces and each faces has two vertices?

Reply:
yes.